internal/transport/transport_test.go

```diff
@@ -2664,3 +2664,87 @@ func TestConnectionError_Unwrap(t *testing.T) {
 		t.Error("ConnectionError does not unwrap")
 	}
 }
+
+// Test that in the event of a graceful client transport shutdown, i.e.,
+// clientTransport.Close(), client sends a goaway to the server with the correct
+// error code and debug data.
+func (s) TestClientSendsAGoAwayFrame(t *testing.T) {
+	// Create a server.
+	lis, err := net.Listen("tcp", "localhost:0")
+	if err != nil {
+		t.Fatalf("Error while listening: %v", err)
+	}
+	defer lis.Close()
+	// greetDone channel verifies that greet is done with the connection
```

Comment:
nit: the channel is "used" to verify. So i might rephrase like
```suggestion
	// greetDone is used to notify when server is done greeting the client.
```

and for the following ones.